### ISSUE TITLE
[4] Setting Up the RL Agent - T[2-4]

### ISSUE THREAD

**Issue by SeanMarcusJensen** [SeanMarcusJensen/ACIT4610_Portifolio_KMHS#24]
**Is your feature request related to a problem? Please describe.**
Setting Up the RL Agent:
• Implement a basic Q-learning algorithm.
• Initialize the Q-table and implement the learning loop.

Training the RL Agent:
• Train the RL agent on the Taxi-v3 environment.
• Tune hyperparameters such as learning rate, discount factor, and exploration
strategy (ε-greedy).

Evaluation:
• Evaluate the performance of the trained RL agent.
• Plot the cumulative rewards over time.
• Compare the performance with a random policy and a heuristic-based policy.

**Describe the solution you'd like**
None.

**Describe alternatives you've considered**
None.

**Additional context**
*We need to establish what kind of framework to use, I recommend Tensorflow*

**Definition of Done**
- [ ] Branch name is `feature/{ISSUE_NUMBER}-{SHORT_DESCRIPTION}`.
- [ ] The code runs without failures. (You didn't break any code).
- [ ] If any new libraries are included, it is added to requirements.txt file.
- [ ] You've updated the `README.md` with the changes done.
- [ ] The `Pull Request` is merged and branch deleted.
